src/Map.ts

```diff
@@ -0,0 +1,261 @@
+import { hasClass } from './decorators';
+import {is} from './object';
+import global from './global';
+
+module Native {
+	interface NativeMap<K, V> {
+		size: number;
+		clear(): void;
+		delete(key: K): boolean;
+		entries(): any;
```

Comment:
The native `Map#entries()` actually returns an iterator and the items are of type `[K, V]`.